packages/realm-web/src/App.test.ts

```diff
@@ -0,0 +1,211 @@
+import { expect } from "chai";
```

Comment:
I would argue not. As these are unit-tests, its best to have them stored as close to the unit they are testing as possible. This pattern becomes significantly more valuable once you have a hierarchy of files that you want to write tests for, as you don't have to maintain two directory tree structures (one in ./src and one in ./tests).

Reply:
Well it might be or might be not. I think having tests in separate directory is more common in modern js libraries. The thing is realm-js has already has opted to follow having tests in separate directory with its own package.json for the tests. I don't think its good to diverge on this and write every new npm package differently.

Reply:
I acknowledge that this is not how we've written unit tests historically in this repository and at the same time I don't think that in itself is reason enough to resist trying a different approach that has advantages (at least in my experience).

If you feel strongly for a seperate ./tests directory, I i won't object against spending moving them - but I think it will introduce a couple of conflicts for me.

Reply:
I think we should be having the tests into separate tests directory. The tests does not need to follow the src directory structure. For example in realm-js package we have split tests per functionality ie tests for realm in realm-tests.js objects in object-tests.js and so on.

If you think this will cause much trouble right now. Could we move them later when you have all other branches based on this one merged into it?

Reply:
Let's postpone that to a follow-up PR then. I do however think theres a subtle difference between these unit-oriented tests and the feature-oriented tests that Realm JS. I agree that feature and integration tests should be located in a seperate directory, but these tests are highly coupled to the unit they're testing, hence the need to be co-located in the directory structure.

Reply:
The existing realm js tests are the same these tests. I don't think there is a good reason to have any tests right besides the source code of the package. In fact it may complicate things. I think this is the reason most frameworks choose to have tests in different directories. 
If this is causing a problem right now lets leave it for a followup PR.